src/apps/vpn/controller.cpp

```diff
@@ -576,7 +578,15 @@ void Controller::handshakeTimeout() {
 
 void Controller::disconnected() {
   logger.debug() << "Disconnected from state:" << m_state;
+  mozilla::glean::session::session_end.set();
 
+  // TODO: This must be after submission of ping - ensure it comes after the submissino
+
+  // We never expect to see this UUID in Glean metrics, as it is also
+  // updated before the next session begins. Rotating it here is a
+  // safety measure.
+  QString sessionId = mozilla::glean::session::session_id.generateAndSet();
```

Comment:
You don't use `sessionId`. Don't create it.

Reply:
We're going to use it in https://mozilla-hub.atlassian.net/browse/VPN-4038, coming shortly after this. Given that this is all building on each other, I have a slight preference towards leaving this variable for now. But if you feel strongly, I'll happily remove.